### PR TITLE
Allowing energy weapons and chainswords to slice apart regular walls

### PR DESCRIPTION
**Pull request by Alberyk**
All energy weapons and chainswords can slice apart regular walls, as suggest in the forums, sometime ago.

### CODE REVIEW

**Comment by NanakoAC** [comment]
I think chainswords should take twice as long as energy blades to do it.

Also, you don't sem to have updated airlocks ?Being able to cut through them is even more important than walls.

My recommendation is 3 seconds for a normal airlock, 10 seconds for a high security/external airlock, and +8 seconds if it's bolted, as there's more places to cut

**Comment by Alberyk** [comment]
You can already crush airlocks with things, however.

File: code/game/objects/structures/girders.dm [hunk 1]
```diff
@@ -76,6 +76,37 @@
...
+		if(do_after(user,30))
+			if(!src) return
+			user << "<span class='notice'>You slice apart the girder!</span>"
+			dismantle()
+			
+	else if(istype(W, /obj/item/weapon/melee/chainsword))
+		var/obj/item/weapon/melee/chainsword/WT = W
+		if(WT.active)
+			user << "<span class='notice'>Now slicing apart the girder...</span>"
+			if(do_after(user,30))
```
Comment (skull132) [inline code/game/objects/structures/girders.dm]: This needs to be longer than the energy swords.

File: code/game/objects/structures/girders.dm [hunk 2]
```diff
@@ -76,6 +76,37 @@
...
+				if(!src) return
+				user << "<span class='notice'>You slice apart the girder!</span>"
+				dismantle()
+			else
+				user << "<span class='notice'>You need to activate the weapon to do that!</span>"
+				return
+			
+	else if(istype(W, /obj/item/weapon/melee/energy/blade))
```
Comment (skull132) [inline code/game/objects/structures/girders.dm]: This isn't activated by the above check? I thought `istype(W, /obj/item/weapon/melee/energy)` returns true if W is a subclass of that parent as well?

File: code/game/objects/structures/girders.dm (hunk 1 above)
Comment (Alberyk) [inline code/game/objects/structures/girders.dm]: There, should be 2x longer.

File: code/game/objects/structures/girders.dm (hunk 2 above)
Comment (Alberyk) [inline code/game/objects/structures/girders.dm]: Fix'd as well.

**Pull request merged**